src/Sentry/Internal/Http/HttpTransport.cs

```diff
@@ -220,11 +226,19 @@ public async Task SendEnvelopeAsync(Envelope envelope, CancellationToken cancell
                         responseString
                     );
                 }
+
+                // If debug level, dump the whole envelope to the logger
+                if (_options.DiagnosticLogger?.IsEnabled(SentryLevel.Debug) is true)
+                {
+                    _options.DiagnosticLogger?.LogDebug("Failed envelope '{0}' has payload:\n{1}",
```

Comment:
maybe indicate when the payload start and where it ends, just to not mix with other logs.

```suggestion
                    _options.DiagnosticLogger?.LogDebug("Failed envelope '{0}' has payload:\n Payload star\n{1}\n Payload end.",
```

Reply:
There's a typo on `start` -> `star` so I can't 'just accept' your suggestion.
Also there's a space after the last line break.
```suggestion
                    _options.DiagnosticLogger?.LogDebug("Failed envelope '{0}' has payload:\n{1}\n",
```